source/mainnet/smart-contracts/tutorials/wCCD/wCCD-front-end-set-up.rst

```diff
@@ -4,6 +4,354 @@
+    will guide you through these setup steps.
+
+    These prerequisites ensure that you have a testnet node reachable locally on port 10001. The browser wallet requires a
+    `JSON-RPC server <https://github.com/Concordium/concordium-json-rpc>`_
+    that points to your node. Clone the repository with the command:
+
+    .. code-block:: console
+
+        $git clone https://github.com/Concordium/concordium-json-rpc.git
+
+    Build and run the server as described in the README file of the
+    `JSON-RPC repo <https://github.com/Concordium/concordium-json-rpc>`_.
+
+    The final command that you execute to start the ``JSON-RPC server`` is as follows:
+
+    .. code-block:: console
+
+        $yarn start --port 9095 --nodeAddress 127.0.0.1 --nodePort 10001 --nodeTimeout 5000
+
+    Your ``JSON-RPC server`` is running on port `9095` and connects to your local node on port `10001`.
+    You are ready now to build the pre-MVP browser wallet from the source code.
+    Clone the repository with the command:
+
+    .. code-block:: console
+
+        $git clone https://github.com/Concordium/concordium-browser-wallet.git
+
+    You can build the browser wallet extension and load it
+    (``dist`` folder from the path `root/packages/browser-wallet`) into the
+    `Chrome browser <https://developer.chrome.com/docs/extensions/mv3/getstarted/#unpacked>`_ as
+    described in the README file of the
+    `browser wallet repo <https://github.com/Concordium/concordium-browser-wallet/tree/main/packages/browser-wallet>`_.
+
+    .. note::
+
+        Depending on the exact commit hash that you used to build your pre-MVP browser wallet, the
+        screenshots and setup steps might differ. The browser wallet hosts the private keys corresponding
+        to the accounts of the user and a link that points to a ``JSON-RPC server``.
+
+    The next steps are based on the pre-MVP browser wallet from a git commit before around 22.8.2022.
+    These earlier versions of the pre-MVP browser wallet have an input field for the private key
+    to import an already existing account and an input field for the link to the ``JSON-RPC server``.
+    Alternatively, you can follow workflow 2 to use the most recent pre-MVP browser wallet
+    without hosting your own node and ``JSON-RPC server``.
+
+    You are ready now to start the browser wallet by clicking on the Concordium icon at the top right of the
+    ``Chrome`` browser.
+
+    .. image:: ./images/wCCD_tutorial_12.png
+        :width: 100 %
+
+    .. note::
+
+        The puzzle icon at the top right of the ``Chrome`` browser allows you to manage your browser extensions.
+        You can enable pinning of the Concordium browser wallet.
+
+        .. image:: ./images/wCCD_tutorial_13.png
+            :width: 30 %
+
+    .. dropdown:: Getting your private key from an account already imported to the `concordium-client`
```

Comment:
I think there is a bit too much nesting going on, which makes the flow less clear.
Perhaps you can expand https://developer.concordium.software/en/mainnet/net/references/developer-tools.html instead and link to it? That might be useful for other guides/tutorials as well.

Reply:
moved it to developer-tools guide and now link to it.